frontend/class-frontend.php

```diff
@@ -1876,4 +1879,20 @@ private function is_premium() {
 		return file_exists( WPSEO_PATH . 'premium/' );
 	}
 
+	/**
+	 * Getting the keywords
+	 *
+	 * @param array|WP_Post $post
+	 *
+	 * @return string
+	 */
+	private function get_keywords( $post )  {
+		$keywords = WPSEO_Meta::get_value( 'metakeywords', $post->ID );
+		if ( $keywords === '' && ( is_object( $post ) && ( isset( $this->options[ 'metakey-' . $post->post_type ] ) && $this->options[ 'metakey-' . $post->post_type ] !== '' ) ) ) {
```

Comment:
`isset( $this->options[ 'metakey-' . $post->post_type ] ) && $this->options[ 'metakey-' . $post->post_type ] !== ''` is practically equivalent to more succinct `! empty( $this->options[ 'metakey-' . $post->post_type ] )`.

Also I would assign complicated key to a variable if used multiple times.